### PR TITLE
Change subprocess.exit_status to subprocess.exitCode

### PR DESCRIPTION
**Pull request by daviditen**
Change subprocess.exit_status to subprocess.exitCode

Added a deprecation warning for subprocess.exit_status. Added a deprecated
test for subprocess.exit_status.

Signed-off-by: David Iten <daviditen@users.noreply.github.com>

### CODE REVIEW

**Comment by daviditen** [comment]
Passed standard paratest.

File: test/c2chapel/c2chapel-tester.chpl [hunk 1]
```diff
@@ -4,7 +4,7 @@ var testdir = CHPL_HOME + "/tools/c2chapel/test/";
 var command = "cd %s; ./tester.sh".format(testdir);
 var sub = Spawn.spawnshell(command);
 sub.communicate();
-if sub.exit_status==0 then
+if sub.exitCode==0 then
```
Comment (arezaii) [inline test/c2chapel/c2chapel-tester.chpl:7]: just a very small nit: there is inconsistent use of spaces around `==`, sometimes it is`var==val` and others it is `var == val`. This is a thing throughout the codebase as far as I can tell, so probably not a big deal to ignore it here. I'm not even certain we have a style guide for this, but the preference _seems_ to be to include the spaces around `==`.

**Review by arezaii — APPROVED** [review]
The changes all look good to me. I haven't considered if there were _other_ places that would need changing, but I assume you did and that this is all of them :) I made one comment about spacing around `==`, whether you change it or not is up to you.

File: test/c2chapel/c2chapel-tester.chpl (hunk 1 above)
Comment (daviditen) [reply test/c2chapel/c2chapel-tester.chpl:7]: > I haven't considered if there were other places that would need changing, but I assume you did and that this is all of them

This is all of them that show up when running our full test suite, so hopefully it's good.

> there is inconsistent use of spaces around `==`, sometimes it is `var==val` and others it is `var == val`

My preference is to include the spaces, but I avoided making any changes here that were unrelated to the renaming.

Thanks!

**Comment by daviditen** [comment]
> Note that arkouda uses exit_status here.

https://github.com/Bears-R-Us/arkouda/pull/916

**Pull request merged**